src/builder/create_attachment.rs

```diff
@@ -117,4 +106,178 @@ impl CreateAttachment {
+    ///
+    /// If you only want to keep a subset of attachments from the message, either implement this
+    /// method manually, or use [`Self::remove()`].
+    ///
+    /// **Note: this EditAttachments must be run on the same message as is supplied here, or else
+    /// Discord will throw an error!**
+    pub fn keep_all(msg: &Message) -> Self {
+        Self {
+            new_and_existing_attachments: msg
+                .attachments
+                .iter()
+                .map(|a| {
+                    NewOrExisting::Existing(ExistingAttachment {
+                        id: a.id,
+                    })
+                })
+                .collect(),
+        }
+    }
+
+    /// This method adds an existing attachment to the list of attachments that are kept after
+    /// editing.
+    ///
+    /// Opposite of [`Self::remove`].
+    pub fn keep(mut self, id: AttachmentId) -> Self {
+        self.new_and_existing_attachments.push(NewOrExisting::Existing(ExistingAttachment {
+            id,
+        }));
+        self
+    }
+
+    /// This method removes an existing attachment from the list of attachments that are kept after
+    /// editing.
+    ///
+    /// Opposite of [`Self::keep`].
+    pub fn remove(mut self, id: AttachmentId) -> Self {
+        #[allow(clippy::match_like_matches_macro)] // `matches!` is less clear here
+        self.new_and_existing_attachments.retain(|a| match a {
+            NewOrExisting::Existing(a) if a.id == id => false,
+            _ => true,
+        });
+        self
+    }
+
+    /// Adds a new attachment to the attachment list.
+    #[allow(clippy::should_implement_trait)] // Clippy thinks add == std::ops::Add::add
+    pub fn add(mut self, attachment: CreateAttachment) -> Self {
+        self.new_and_existing_attachments.push(NewOrExisting::New(attachment));
+        self
+    }
+
+    /// Clones all new attachments into a new Vec, keeping only data and filename, because those
+    /// are needed for the multipart form data. The data is taken out of `self` in the process, so
+    /// this method can only be called once.
+    pub(crate) fn take_files(&mut self) -> Vec<CreateAttachment> {
+        let mut id_placeholder = 0;
+
+        let mut files = Vec::new();
+        for attachment in &mut self.new_and_existing_attachments {
+            if let NewOrExisting::New(attachment) = attachment {
+                let mut cloned_attachment = CreateAttachment::bytes(
+                    std::mem::take(&mut attachment.data),
+                    attachment.filename.clone(),
+                );
+
+                // Assign placeholder IDs so Discord can match metadata to file contents
+                attachment.id = id_placeholder;
+                cloned_attachment.id = id_placeholder;
+                files.push(cloned_attachment);
+
+                id_placeholder += 1;
+            }
+        }
+        files
+    }
+
+    pub(crate) fn is_empty(&self) -> bool {
```

Comment:
```suggestion
    #[cfg(feature = "cache")]
    pub(crate) fn is_empty(&self) -> bool {
```
Get an unused warning if cache is disabled.